### PR TITLE
[cdc-base] Close idle readers when snapshot finished

### LINKED ISSUE #1731: [cdc-base] Close idle readers when snapshot phase ended.
**Is your feature request related to a problem? Please describe.**
Before in Flink 1.14, checkpoint on finished source feature is not supported.
We can not close the idle readers when parallel snapshot finished.

**Describe the solution you'd like**
If the runtime Flink version is over 1.14, close idle readers when snapshot phase ended to reduce resource usage.

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
Close idle readers when snapshot finished #2162.

This will fix #1731.

### CODE REVIEW

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceOptions.java [hunk 1]
```diff
@@ -245,4 +245,15 @@ public class MySqlSourceOptions {
...
                                     + "By default, the chunk key is the first column of the primary key."
                                     + "This column must be a column of the primary key.");
+
+    @Experimental
+    public static final ConfigOption<Boolean> SCAN_INCREMENTAL_CLOSE_IDLE_READER_ENABLED =
+            ConfigOptions.key("scan.incremental.close-idle-reader.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Whether to close idle readers at the end of the snapshot phase. This feature depends on "
```
Comment (ruanhang1993) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceOptions.java:255]:
```
"Whether to close idle readers at the end of the snapshot phase. This feature depends on "
  + "FLIP-147: Support Checkpoints After Tasks Finished. The flink version is required to be "
   + "greater than or equal to 1.14 when enabling this feature."
```

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/enumerator/MySqlSourceEnumerator.java [hunk 2]
```diff
@@ -193,6 +194,12 @@ private void assignSplits() {
                 context.assignSplit(mySqlSplit, nextAwaiting);
                 awaitingReader.remove();
                 LOG.info("Assign split {} to subtask {}", mySqlSplit, nextAwaiting);
+            } else if (isAssigningFinished(splitAssigner.getAssignerStatus())
```
Comment (ruanhang1993) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/enumerator/MySqlSourceEnumerator.java:197]: See the comment in `IncrementalSourceEnumerator` at the same part.

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Review by GOODBOY008 — APPROVED** [review]
LGTM

**Pull request merged**